Added read functionality for FAT32

DrDeano opened the pull request:
Create test FAT32 image on build
Moved test files to test_files directory
Removed global array for test fat32
Part of #216

SamTebbs33 commented on src/kernel/filesystem/fat32.zig:
```diff
@@ -3837,87 +3753,195 @@ test "Fat32FS.openImpl - entry iterator failed 2nd next" {
+
+    var buff = [_]u8{0xAA} ** 8450;
+    const read = try test_node.read(buff[0..]);
+    expectEqual(read, 8450);
+    // The buffer should have:
+    // A ** 64\n
+    // BBB..
+    // CCC..
+    // .....
+    // ZZZ..
+    // 5 times
+    var i: usize = 0;
+    // Repeats 5 times
+    while (i < 5) : (i += 1) {
+        // 26 characters
+        var j: usize = 0;
+        while (j < 26) : (j += 1) {
+            // Each character repeated 64 times
+            var k: usize = 0;
+            while (k < 64) : (k += 1) {
+                expectEqual(buff[(i * 65 * 26) + (j * 65) + k], 'A' + @intCast(u8, j));
+            }
+            // New line
+            expectEqual(buff[(i * 65 * 26) + (j * 65) + 64], '\n');
+        }
+    }
```
Instead of this, you could use `@embedFile("test/fat32/test_files/large_file2.txt")` to get a slice of the file's contents and do a slice equality check. It's up to you depending on if you think it's cleaner or not!

DrDeano replied:
That is a good idea

SamTebbs33 reviewed (approved):
Looking very good. I like the breadth of testing. I just have one small suggestion but it's up to you.

DrDeano committed:
Added read functionality to FAT32

Create test FAT32 image on build
Moved test files to test_files directory
Removed global array for test fat32
Part of #216
Removed hard coded test files


Use @embedFile for large file test
